File: src/main/java/com/silverboxsoft/dynamodbtool/controller/inputdialog/DynamoDbBinarySetInputDialog.java
package com.silverboxsoft.dynamodbtool.controller.inputdialog;

import java.util.List;

import com.silverboxsoft.dynamodbtool.utils.DynamoDbUtils;

import javafx.scene.Node;
import javafx.scene.control.TextField;
import software.amazon.awssdk.core.SdkBytes;

public class DynamoDbBinarySetInputDialog extends AbsDynamoDbSetInputDialog<SdkBytes> {

	public DynamoDbBinarySetInputDialog(List<SdkBytes> dynamoDbRecord, String dialogTitle) {
		super(dynamoDbRecord, dialogTitle);
	}

	@Override
	protected String getTypeString() {
		return "BINARY";
	};

	@Override
	protected Node getAttrubuteBox(int recIndex, SdkBytes attr) {
		TextField textField = new TextField();
		textField.setText(DynamoDbUtils.getBase64StringFromSdkBytes(attr));
		return textField;
	}

	@Override
	protected SdkBytes getCurrentAttrubuteValue(Node valueNode) {
		TextField valField = (TextField) valueNode;
		return DynamoDbUtils.getSdkBytesFromBase64String(valField.getText());
	}

	@Override
	protected SdkBytes getInitAttribute() {
		return SdkBytes.fromByteArray(new byte[0]);
	}

	@Override
	protected boolean isSameValue(SdkBytes valA, SdkBytes valB) {
		return valA.equals(valB);
	}
}

File: src/main/java/com/silverboxsoft/dynamodbtool/controller/inputdialog/DynamoDbBinarySetInputDialog.kt
package com.silverboxsoft.dynamodbtool.controller.inputdialog

import com.silverboxsoft.dynamodbtool.utils.DynamoDbUtils
import software.amazon.awssdk.core.SdkBytes
import javafx.scene.Node
import javafx.scene.control.*

class DynamoDbBinarySetInputDialog(dynamoDbRecord: List<SdkBytes>, dialogTitle: String) : AbsDynamoDbSetInputDialog<SdkBytes>(dynamoDbRecord, dialogTitle) {
    override val typeString: String
        get() = "BINARY"

    override fun getAttributeBox(recIndex: Int, attr: SdkBytes): Node {
        val textField = TextField()
        textField.text = DynamoDbUtils.Companion.getBase64StringFromSdkBytes(attr)
        return textField
    }

    override fun getCurrentAttributeValue(valueNode: Node): SdkBytes {
        val valField = valueNode as TextField
        return DynamoDbUtils.Companion.getSdkBytesFromBase64String(valField.text)
    }

    override val initialAttribute: SdkBytes
        get() = SdkBytes.fromByteArray(ByteArray(0))

    override fun isSameValue(valA: SdkBytes, valB: SdkBytes): Boolean {
        return valA == valB
    }
}